modules/extensions/src/clip/clip-extension.ts

```diff
@@ -77,9 +86,12 @@ in float clip_isVisible;
  * The fragment-shader version clips pixels at the bounds
  * e.g. PolygonLayer - show the part of the polygon that intersect with the bounds
  */
-const shaderModuleFs: ShaderModule = {
-  name: 'clip-fs',
-  fs: shaderFunction
+const shaderModuleFs: ShaderModule<ClipModuleSettings> = {
+  name: 'clip',
```

Comment:
Any risk sharing the name between the modules?

Reply:
> Any risk sharing the name between the modules?

I am not quite sure what the context is but I would avoid using same name for two modules if that is the question.

Reply:
[Luma assumes](https://github.com/visgl/luma.gl/blob/4e28a224d0f2f7990ee107bfa52044e09106a68b/modules/engine/src/model/model.ts#L519) that the uniform block will be called `${moduleName}Uniforms` - which for starters means we cannot use `clip-fs`. 

It also means that if we cannot share the same uniformBlock across multiple modules. Perhaps we should add an optional `ShaderModule.uniformBufferName` in luma?